experimental/algorithm/LAGraph_MTX_reach_basic.c

```diff
@@ -0,0 +1,116 @@
+#define LG_FREE_WORK                                                                     \
+    do {                                                                                 \
+        for (size_t i = 0; i < T_size; i++) {                                            \
+            GrB_free(&T[i]);                                                             \
+        }                                                                                \
+        free(nnz);                                                                       \
+    } while (0)
+
+#include "LG_internal.h"
+#include <LAGraphX.h>
+
+GrB_Info LAGraph_MTX_reach_basic(GrB_Matrix *output, GrB_Matrix *adj_matrices,
```

Comment:
1. Comment with description is required. 
   - General description (problem statement)
   - Input and output description with implicit contracts description: all numbers starts from 0, correspondence between matrices and terminal numbers, etc.    
2. Name is not good. It is context-free reachability algorithm algorithm. Not just reachability.